lucet-concurrency-tests/tests/killswitch.rs

```diff
@@ -0,0 +1,114 @@
+use lucet_runtime::{Error, KillSuccess, Limits, Region, TerminationDetails};
+use std::sync::Arc;
+use std::thread;
+
+use lucet_module::FunctionPointer;
+use lucet_runtime::MmapRegion;
+use lucet_runtime_internals::lock_testpoints::Syncpoint;
+use lucet_runtime_internals::module::Module;
+use lucet_runtime_internals::module::{MockExportBuilder, MockModuleBuilder};
+use lucet_runtime_internals::vmctx::lucet_vmctx;
+
+static mut ENTERING_GUEST: Option<Syncpoint> = None;
+
+pub fn mock_traps_module() -> Arc<dyn Module> {
```

Comment:
This is lifted pretty directly from `timeouts.rs`, and includes functionality enough that I think it's sufficient for here too.